src/main/scala/cromwell/engine/backend/local/SharedFileSystem.scala

```diff
@@ -101,102 +111,94 @@ trait SharedFileSystem {
-   * of the workflow execution directory.  Return a Map of the input values with any input WdlFiles adjusted to
-   * reflect host paths.
-   *
-   * Original input path: /could/be/anywhere/input.bam
-   * Host inputs path: $PWD/cromwell-executions/some-workflow-name/0f00-ba4/workflow-inputs/input.bam
+   * Try to localize a WdlValue if it is or contains a WdlFile.
+   * @param toDestPath function specifying how to generate the destination path from the source path
+   * @param strategies strategies to use for localization
+   * @param postProcessPath optional function to be applied to the path after the file it points to has been localized (defaults to noop)
+   * @param wdlValue WdlValue to localize
+   * @return localized wdlValue
    */
-  private def stageWorkflowInputs(descriptor: WorkflowDescriptor): Try[HostInputs] = {
-    val hostInputsPath = Paths.get(LocalBackend.hostExecutionPath(descriptor).toFile.getAbsolutePath, "workflow-inputs")
-    val attemptedStagedInputs = descriptor.actualInputs map { case(name, value) =>
-      val call = descriptor.namespace.resolve(name.split("\\.").dropRight(1).mkString(".")) match {
-        case Some(c: Call) => Some(c)
-        case _ => None
-      }
-      name -> stageWdlValue(call, value, hostInputsPath)
-    }
-    attemptedStagedInputs.values collect { case f: Failure[_] => f } match {
-      case Nil =>
-        Success(attemptedStagedInputs map { case (k, v) => k -> v.get })
-      case failedInputs =>
-        val errors = failedInputs map { _.exception.getMessage } mkString "\n"
-        Failure(new UnsupportedOperationException(s"Failures during localization:\n\n$errors"))
+  def localizeWdlValue(toDestPath: (Path => Try[Path]), strategies: Stream[LocalizeStrategy], postProcessPath: Option[Path => Path] = None)(wdlValue: WdlValue): Try[WdlValue] = {
+
+    def localize(source: Path, dest: Path) = strategies map { _(source, dest) } find { _.isSuccess } getOrElse {
+      Failure(throw new UnsupportedOperationException(s"Could not localize $source -> $dest"))
     }
-  }
 
-  private def stageInputArray(call: Option[Call], array: WdlArray, hostInputsPath: Path): Try[WdlArray] = {
-    val attemptedStagedArray = array.value.map(stageWdlValue(call, _, hostInputsPath))
-    attemptedStagedArray collect { case f: Failure[_] => f } match {
-      case s: Seq[Failure[_]] if s.nonEmpty =>
-        val errors = s map { _.exception.getMessage } mkString "\n"
-        Failure(new UnsupportedOperationException(s"Failures during localization of array $array:\n\n$errors"))
-      case _ => Success(WdlArray(array.wdlType, attemptedStagedArray.map(_.get)))
+    def adjustArray(t: WdlArrayType, inputArray: Seq[WdlValue]): Try[WdlArray] = {
+      val tryAdjust = inputArray map localizeWdlValue(toDestPath, strategies, postProcessPath)
+
+      TryUtil.flatten(tryAdjust, s"Failed to localize files in input Array ${wdlValue.valueString}") map { adjusted =>
```

Comment:
Ha ok, I think we wrote the same function separately on 2 PRs so I just merged them, but i didn't know there had been a consensus on `sequence`. I'll rename it.